### ISSUE TITLE
glb files - too dark normal maps after import

### ISSUE THREAD

**Issue by sebastianwibergmicrosoft** [KhronosGroup/UnityGLTF#362]
![correct](https://user-images.githubusercontent.com/43607781/52958565-9b963680-3394-11e9-8cfc-0b5d66eb1c24.png)
![dark](https://user-images.githubusercontent.com/43607781/52958573-9fc25400-3394-11e9-9c2d-08ade0917412.png)

File imports properly in other applications.
In UnityGLTF the file has 118, 118, 255 as "neutral" normal color, instead of 127,127,255.
This completely breaks some models, that rely on normal maps to describe geometry.

**Comment by sebastianwibergmicrosoft** [comment]
I did some research.
The buffers are correct, but when using "EncodeToPNG" the texture becomes to dark.
I did a test inside ConstructUnityTexture, I wrote the raw bufer to a png file, then did texture.LoadImage(...) then texture.Apply() and finally texture.EncodeToPNG. This texture ended up too dark (like illustrated above).

**Comment by blgrossMS** [comment]
This sounds like it could be an srgb issue. Does changing the Unity value from gamma to linear fix the issue?

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by sebastianwibergmicrosoft** [comment]
It isnt a big enough change to be srgb. A conversion of that kind should put it at 90, not 118.
Also, I did the write, loadimage, encode, write directly after one another, all using linear. There should be no conversion. I have seen some threads on the unity forums about encodeToPng being a nit dark as well.
A solution, from our end, is to not use the encoder in unity, and instead write the raw buffer and use that. I tried and it worked perfectly.
